trl/trainer/grpo_trainer.py

```diff
@@ -839,22 +846,34 @@ def _generate_and_score_completions(
         self._metrics[mode]["reward_std"].append(std_grouped_rewards.mean().item())
 
         if (
-            self.log_completions
-            and self.state.global_step % self.args.logging_steps == 0
-            and "wandb" in self.args.report_to
+            self.log_completions_steps is not None
+            and self.log_completions_steps > 0
+            and self.state.global_step % self.log_completions_steps == 0
```

Comment:
why do you replace this

```diff
- self.state.global_step % self.args.logging_steps == 0
+ self.state.global_step % self.log_completions_steps == 0
```

logging steps seems sufficient here

Reply:
The idea was to decouple `log_completions_steps` from `logging_steps` since the latter could lead to a lot prints to the console. The user may not want to see a big table of text every 10 seconds.

We could bind it to `logging_steps` again if you think it's not necessary

Reply:
Yes I think it's not necessary, `logging_step` seems necessary. We could still decouple them later if it's raelly to much noise in the log

Reply:
Alright, I reversed it. Now it should also be compatible with the old API.